custom_components/mqtt_vacuum_camera/utils/connection/decompress.py

```
@@ -0,0 +1,452 @@
+    @staticmethod
+    @lru_cache(maxsize=32)
+    def get_instance() -> "DecompressionManager":
+        """Get the singleton instance of DecompressionManager.
+        This method is cached to avoid creating multiple instances.
+
+        Returns:
+            The singleton instance
+        """
+        return DecompressionManager()
+
+    @lru_cache(maxsize=32)
```

Ruff (0.8.2)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)